play-json/shared/src/main/scala/Reads.scala

```diff
@@ -398,8 +398,57 @@ trait DefaultReads extends LowPriorityDefaultReads {
   /**
    * Deserializer for Array[T] types.
    */
-  implicit def ArrayReads[T: Reads: ClassTag]: Reads[Array[T]] = new Reads[Array[T]] {
-    def reads(json: JsValue) = json.validate[List[T]].map(_.toArray)
+  implicit def ArrayReads[T: Reads: ClassTag]: Reads[Array[T]] =
+    Reads[Array[T]] { _.validate[List[T]].map(_.toArray) }
+
+  // If the number is decimal, considering it's /1000 of millis,
+  // otherwise considering it's directly in millis
+  private def numberDuration(n: BigDecimal): FiniteDuration = {
+    val millis = if (!n.ulp.isWhole) {
+      (n.toDouble * 1000).toLong
+    } else n.toLong
+
+    FiniteDuration(millis, "ms")
+  }
+
+  /**
+   * Deserializer for Scala FiniteDuration
+   */
+  implicit val finiteDurationWrites: Reads[FiniteDuration] =
+    Reads[FiniteDuration] {
+      case JsNumber(n) => JsSuccess(numberDuration(n))
+      case JsString("0") => JsSuccess(Duration.Zero)
+      case JsString("Undefined") => JsError("error.invalid.finiteDuration")
+      case JsString(input) => try {
+        val d = Duration(input)
+
+        if (d.isFinite) JsSuccess(FiniteDuration(d.length, d.unit))
+        else JsError("error.invalid.finiteDuration")
+      } catch {
+        case _: Throwable => JsError("error.invalid.duration")
```

Comment:
Even if only `NumberFormatException` should be possible for now, for me that's safe to handle any exception there.